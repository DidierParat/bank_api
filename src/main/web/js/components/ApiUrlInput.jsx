import React, { Component } from 'react';
import { connect } from 'react-redux';
import { setApiUrl } from '../data/dataStore';

export class ApiUrlInput extends Component {
    constructor(props) {
        super(props);
        this.handleChange = this.handleChange.bind(this);
    }

    handleChange(event) {
        this.props.onChangeApiUrl(event.target.value);
    }

    render() {
        return (
            <p>
                <label>API URL:</label>
                <input defaultValue={this.props.apiUrl} onChange={this.handleChange} />
            </p>
        );
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(ApiUrlInput);

function mapStateToProps(state) {
    return {
        apiUrl: state.apiUrl
    };
}

function mapDispatchToProps(dispatch) {
    return {
        onChangeApiUrl: apiUrl => dispatch(setApiUrl(apiUrl))
    }
}
