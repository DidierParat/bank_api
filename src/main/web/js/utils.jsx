import React from 'react';
import { get } from 'axios';

export function fetchAndSetData(request, setData) {
    get(request)
        .then(response => setData(response.data))
        .catch(error => console.error(error));
}