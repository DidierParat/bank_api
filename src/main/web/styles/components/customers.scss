input {
    width: 220px;
    padding-left: 5px;
    margin-left: 10px;
}