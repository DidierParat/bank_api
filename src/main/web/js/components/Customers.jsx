import React, { Component } from 'react';
import { fetchAndSetData } from '../utils';
import { connect } from 'react-redux';
import { setCustomersApiData } from '../data/dataStore';

export class Customers extends Component {
    constructor(props) {
        super(props);
        this.handleClick = this.handleClick.bind(this);
    }

    handleClick(event) {
        this.props.setData({});
        fetchAndSetData(
            this.props.apiUrl + "/customers/12345678901",
            this.props.setData);
    }

    render() {
        return (
            <div>
                <h2>Customers</h2>
                <p>GET request: /customers/(customerID)</p>
                <p>Get customer details by ID.</p>
                <button onClick={this.handleClick}>Try !</button>
                <pre>{JSON.stringify(this.props.data, null, 2)}</pre>
            </div>
        );
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(Customers);

function mapStateToProps(state) {
    return {
        apiUrl: state.apiUrl,
        data: state.customersApiData
    };
}

function mapDispatchToProps(dispatch) {
    return {
        setData: data => dispatch(setCustomersApiData(data))
    }
}