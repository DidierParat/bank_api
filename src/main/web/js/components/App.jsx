import React, { Component } from 'react';
import ApiUrlInput from './ApiUrlInput';
import Customers from './Customers';

export default class App extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <div>
                <ApiUrlInput />
                <Customers />
            </div>
        );
    }
}