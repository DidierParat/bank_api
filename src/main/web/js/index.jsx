import React from 'react';
import ReactDOM from 'react-dom';
import App from './components/App';
import { getStore } from './data/dataStore';
import { Provider } from 'react-redux';
import '../styles/app.scss';

ReactDOM.render(
    <Provider store={getStore()}>
        <App />
    </Provider>,
    document.getElementById('app')
);