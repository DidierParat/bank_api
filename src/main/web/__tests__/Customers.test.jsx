import React from 'react';
import { Customers } from '../js/components/Customers';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { fetchAndSetData } from '../js/utils';

jest.mock('../js/utils', () => ({
    fetchAndSetData: jest.fn()
}));

configure({ adapter: new Adapter() });

describe('Test <Customers> with wrapping in <Provider>',()=>{
    it('+++ renders correctly', () => {
        const setData = (data)=>{};
        const rendered = renderer.create(
            <Customers apiUrl={'http://localhost:8080'} data={{}} setData={setData} />
        );
        expect(rendered.toJSON()).toMatchSnapshot();
    });

    it('+++ clicking button sets new data', () => {
        const setData = (data)=>{};
        const wrapper = shallow( <Customers apiUrl={'http://localhost:8080'} data={{}} setData={setData} /> );
        wrapper.find('button').simulate('click');
        expect(fetchAndSetData).toBeCalledWith('http://localhost:8080/customers/123', setData);
    });

    it('+++ data is displayed as json', () => {
        const setData = (data)=>{};
        const wrapper = shallow( <Customers apiUrl={'http://localhost:8080'} data={{status: 200}} setData={setData} /> );
        expect(wrapper.find('pre').get(0).props.children).toBe("{\n  \"status\": 200\n}");
    });
});