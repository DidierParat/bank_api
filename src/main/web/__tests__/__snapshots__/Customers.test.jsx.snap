// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <Customers> with wrapping in <Provider> +++ renders correctly 1`] = `
<div>
  <h2>
    Customers
  </h2>
  <p>
    GET request: /customers/(customerID)
  </p>
  <p>
    Get customer details by ID.
  </p>
  <button
    onClick={[Function]}
  >
    Try !
  </button>
  <pre>
    {}
  </pre>
</div>
`;
